anta/tests/interfaces.py

```diff
@@ -438,3 +483,78 @@ def test(self) -> None:
             self.result.is_failure(f"Some L2 interfaces do not have correct MTU configured:\n{wrong_l2mtu_intf}")
         else:
             self.result.is_success()
+
+
+class VerifyInterfacesSpeed(AntaTest):
+    """
+    Verifies the speed, lanes, auto-negotiation status, and mode as full duplex for interfaces.
+    If speed is auto then verify auto-negotiation as success and mode as full duplex.
+    If speed is auto with a value(auto 10g) then verify auto-negotiation as success, mode as full duplex and speed/lanes as per input.
+    If speed is forced with a value(forces 10g) then verify mode as full duplex and speed as per input.
+    If speed with lane(100g-8) then verify mode as full duplex and speed/lanes as per input.
+
+    Expected Results:
+        * Success: The test will pass if an interface is configured with the correct speed, lanes, auto-negotiation and mode as full duplex.
+        * Failure: The test will fail if an interface is not found, the speed or alnes does not match with input,
+                   auto-negotiation is not correct or mode is not full duplex.
+    """
+
+    name = "VerifyInterfacesSpeed"
+    description = "Verifies the speed, lanes, auto-negotiation status, and mode as full duplex for interfaces."
+    categories = ["interfaces"]
+    commands = [AntaCommand(command="show interfaces")]
+
+    class Input(AntaTest.Input):
+        """Inputs for the VerifyInterfacesSpeed test."""
+
+        interfaces: List[Interfaces]
```

Comment:
Can we name `Interfaces` something else? It can be confusing with the custom type `Interface` we are importing.

Reply:
InterfaceDetail